### PR TITLE
keep toCase function the same as golang implementation

### LINKED ISSUE #7695: The result of `upper()` and `lower()` is different with TiDB
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> drop table t;
Query OK, 0 rows affected (0.34 sec)

mysql> create table t(a char(20));
Query OK, 0 rows affected (0.15 sec)

mysql> insert into t values ('𞤀');
Query OK, 1 row affected (0.01 sec)

mysql> ALTER TABLE t set tiflash replica 1;
Query OK, 0 rows affected (0.14 sec)

mysql> set @@session.tidb_isolation_read_engines = 'tiflash';
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t where lower(a) = '𞤀';
+------+
| a    |
+------+
| 𞤀     |
+------+
1 row in set (0.07 sec)

mysql> set @@session.tidb_isolation_read_engines = 'tikv';
Query OK, 0 rows affected (0.01 sec)

mysql> select * from t where lower(a) = '𞤀';
Empty set (0.00 sec)
```

### 2. What did you expect to see? (Required)
```
Same as TiDB and TiKV
```

### 3. What did you see instead (Required)

```
Result not same
```

### 4. What is your TiFlash version? (Required)

2023-06-24 nightly

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #7695

Problem Summary:

### What is changed and how it works?
The core code for TiFlash's `lower/upper` function is
https://github.com/pingcap/tiflash/blob/57c0c57e91a1e95f2ff4a13597a2898409ce11de/dbms/src/Functions/CharUtil.h#L235
And this function is actually a directly translation of golang's implementation:
https://github.com/golang/go/blob/go1.21.3/src/unicode/letter.go#L211

This pr update the code in TiFlash to keep it the same as latest golang's implemenation.
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/fullstack-test/expr/issue_7695.test [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+mysql> alter table test.t set tiflash replica 1
+func> wait_table test t
+
+mysql> insert into test.t values ('𞤀');
+mysql> set tidb_enforce_mpp=1; select a, lower(a) from test.t;
++------+----------+
+| a    | lower(a) |
++------+----------+
+| 𞤀     | 𞤢         |
++------+----------+
```
Comment (windtalker) [inline tests/fullstack-test/expr/issue_7695.test:26]: It should be a display issue, the real result looks like this
Screenshot 2023-10-25 at 17 53 19

**Review by SeaRise — APPROVED** [review]

File: tests/fullstack-test/expr/issue_7695.test (hunk 1 above)
Comment (SeaRise) [reply tests/fullstack-test/expr/issue_7695.test:26]: How about adding unit test instead of Integration  test?

**Review by yibin87 — APPROVED** [review]
LGTM

**Pull request merged**